File: Data Structures & Algorithms/binary-search/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums, target) {

        let i = 0;
        let j = nums.length-1;
        while(i<=j){
            let midIndex = Math.floor((i + j)/2);
            if(nums[midIndex] == target) return midIndex;
            if(target>nums[midIndex]){
                i = midIndex+1
            }else{
                j= midIndex-1;
            }
        }
    return -1;
    }
}

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-0.java
class Solution {
    public int search(int[] nums, int target) {
    int low = 0, high = nums.length - 1;

    while (low <= high) {
        int mid = low + (high - low) / 2;

        if (nums[mid] == target) return mid;

        // Left half is sorted
        if (nums[low] <= nums[mid]) {
            // Target lies within the sorted left half
            if (target >= nums[low] && target < nums[mid]) {
                high = mid - 1;   // go left
            } else {
                low = mid + 1;    // go right
            }
        }
        // Right half is sorted
        else {
            // Target lies within the sorted right half
            if (target > nums[mid] && target <= nums[high]) {
                low = mid + 1;    // go right
            } else {
                high = mid - 1;   // go left
            }
        }
    }

    return -1;
}
}
